tests/validation/tests/v3_api/test_pipeline.py

```diff
@@ -0,0 +1,108 @@
+from .common import get_global_admin_client_and_cluster, create_kubeconfig, \
+    random_test_name, create_project_and_ns, get_project_client_for_token, \
+    random_str, wait_for_condition, get_cluster_client_for_token, \
+    create_connection, WebsocketLogParse, CATTLE_TEST_URL, USER_TOKEN
+import pytest
+import urllib
+import time
+
+namespace = {"p_client": None, "ns": None, "cluster": None,
+             "project": None, "pipeline": None, "pipeline_run": None}
+PIPELINE_TIMEOUT =  600
+PIPELINE_REPO_URL = "https://github.com/rancher/pipeline-example-go.git"
+
+
+@pytest.fixture(scope='module', autouse="True")
+def create_project_client(request):
+    client, cluster = get_global_admin_client_and_cluster()
+    create_kubeconfig(cluster)
+    p, ns = create_project_and_ns(
+        USER_TOKEN, cluster, random_test_name("testworkload"))
+    p_client = get_project_client_for_token(p, USER_TOKEN)
+    namespace["p_client"] = p_client
+    namespace["ns"] = ns
+    namespace["cluster"] = cluster
+    namespace["project"] = p
+
+    def fin():
+        # project delete doesn't remove all ns in test, removing manually
+        client = get_cluster_client_for_token(namespace["cluster"], USER_TOKEN)
+        for ns_x in client.list_namespace(projectId=namespace["project"].id):
```

Comment:
Why do we need this? I see only one namespace get created under create_project_client()

Reply:
The pipeline creates a ns for the example run with a random name.

Reply:
The pipeline creates it's own namespaces to which the artifacts are deployed.